wire-ios-data-model/Tests/MLS/ProteusToMLSMigrationCoordinatorTests.swift

```diff
@@ -60,10 +66,80 @@ class ProteusToMLSMigrationCoordinatorTests: ZMBaseManagedObjectTest {
         super.tearDown()
     }
 
+    // MARK: - Tests
+
+    func testMigrateOrJoinGroupConversations_CallsJoinGroupForNewConversations() async throws {
+        // GIVEN
+        var groupID: MLSGroupID?
+
+        await syncMOC.perform {
+            let selfUser = ZMUser.selfUser(in: self.syncMOC)
+            selfUser.teamIdentifier = UUID()
+
+            let conversation = ZMConversation.insertNewObject(in: self.syncMOC)
+            conversation.teamRemoteIdentifier = selfUser.teamIdentifier
+            conversation.conversationType = .group
+            conversation.messageProtocol = .mixed
+            conversation.mlsGroupID = .random()
+
+            groupID = conversation.mlsGroupID
+        }
+
+        let conversationGroupID = try XCTUnwrap(groupID)
+
+        mockMLSService.conversationExistsMock = { _ in
+            return false
+        }
+
+        // WHEN
+        await sut.migrateOrJoinGroupConversations()
+
+        // THEN
+        XCTAssertTrue(mockMLSService.calls.joinGroup.contains(conversationGroupID), "joinGroup should be called for new conversations")
+    }
+
+    func testMigrateOrJoinGroupConversations_DoesNotCallJoinGroupForExistingConversations() async throws {
+        // GIVEN
+        var groupID: MLSGroupID?
+
+        await syncMOC.perform {
+            let selfUser = ZMUser.selfUser(in: self.syncMOC)
+            selfUser.teamIdentifier = UUID()
+
+            let conversation = ZMConversation.insertNewObject(in: self.syncMOC)
+            conversation.teamRemoteIdentifier = selfUser.teamIdentifier
+            conversation.conversationType = .group
+            conversation.messageProtocol = .mixed
+            conversation.mlsGroupID = .random()
+
+            groupID = conversation.mlsGroupID
+        }
+
+        let conversationGroupID = try XCTUnwrap(groupID)
+
+        mockMLSService.conversationExistsMock = { _ in
+            return true
+        }
+
+        // WHEN
+        await sut.migrateOrJoinGroupConversations()
+
+        // THEN
+        XCTAssertFalse(mockMLSService.calls.joinGroup.contains(conversationGroupID), "joinGroup should not be called for existing conversations")
+    }
+
     // MARK: - UpdateMigrationStatus
 
     func test_UpdateMigrationStatus_StartsMigration_IfNotStartedAndReady() async {
         // Given
+        await syncMOC.perform {
+            let selfUser = ZMUser.selfUser(in: self.syncMOC)
+            selfUser.teamIdentifier = UUID()
+
+            let conversation = ZMConversation.insertNewObject(in: self.syncMOC)
+            conversation.teamRemoteIdentifier = selfUser.teamIdentifier
+        }
```

Comment:
perhaps this can be moved to the `setUp` method, along with the ones below

Reply:
I fixed that as well.